sage.py

```diff
@@ -571,243 +485,159 @@ def getepisodes(action_seq, mcat, plot, debug=False):
             plt.plot(xax_end, yax, 'r', linestyle=(0, (5, 10)))
 
         plt.show()
-    #print('number episodes ', len(episodes_))
     return episodes_
 
-def aggregate_into_episodes(unparse, team_labels, step=150):
-    cols = ['b', 'r', 'g', 'c', 'm', 'y', 'k', 'olive', 'lightcoral', 'skyblue', 'mediumpurple', 'springgreen', 'chocolate', 'cadetblue', 'lavender']
 
+def aggregate_into_episodes(team_alerts, step=150):
     PRINT = False
-    interesting = []
-    # Reorganize data for each attacker per team 
+
+    # Reorganize data for each attacker per team
     team_data = dict()
-    s_t = dict()
-    for tid,team in enumerate(unparse):
-        #attackers = list(set([x[1] for x in team])) # collecting all src ip
-        #attackers.extend(list(set([x[3] for x in team]))) # collection all dst ip
-        #attackers = [x for x in attackers if x not in hostip.keys()] # filtering only attackers
-        
-        
-        host_alerts = dict()
-
-        for ev in team:
-            #print(ev[0])
-            h = ev[7]
-            s = ev[1]
-            d = ev[3]
-            c = ev[9]
-            ts = ev[8]
-            sp = ev[2] if ev[2] != None else 65000 
-            dp = ev[4] if ev[4] != None else 65000
-            signature = ev[5]
-            # Simply respect the source,dst format! (Correction: source is always source and dest alwyas dest!)
-            
-            source, dest, port = -1, -1, -1
-            #print(s, d, sp, dp)
-            #assert sp >= dp
-            
-            source = s #if s not in inv_hostip.keys() else inv_hostip[s]
-            dest = d #if d not in inv_hostip.keys() else inv_hostip[d]
-            # explicit name if cant resolve
-            #port = str(dp) if (dp not in port_services.keys() or port_services[dp] == 'unknown') else port_services[dp]['name'] 
-            
-            # say unknown if cant resolve it
-            port = 'unknown' if (dp not in port_services.keys() or port_services[dp] == 'unknown') else port_services[dp]['name']
+    for tid, team in enumerate(team_alerts):
+        host_alerts = dict()  # (attacker, victim) -> alerts
+
+        for alert in team:
+            # Alert format: (diff_dt, src_ip, src_port, dst_ip, dst_port, sig, cat, host, ts, mcat)
+            src_ip, dst_ip, signature, ts, mcat = alert[1], alert[3], alert[5], alert[8], alert[9]
+            dst_port = alert[4] if alert[4] is not None else 65000
+            # Simply respect the source,dst format! (Correction: source is always source and dest always dest!)
```

Comment:
Here I also don't really understand. It says `Correction: source is always source and dest always dest!`, but there are still the if-checks for (src,dst) and (dst,src). Are they still needed?

Reply:
Maybe this comment is outdated. Problem is that the check is dataset-dependent. For the datasets where we know attacker IPs, we can convert them into attacker,victim format (e.g., for cptc-2018), but for other datasets, we simply have to believe that src,dst == attacker,victim. I think make_condensed_data() is the function that's doing the real check based on IP format. You could, in theory, move that check in this function already (but make sure to check whether the dataset is cptc-201X, otherwise we get no AGs).

Reply:
We can discuss it during the meeting, looking at a related comment later in the code. Also, there is no easy way to check for the dataset. We can check the directory name, but that can also be weirdly named (I think during RP I just had `cptc/2017`, but someone might also have `alerts/2017` if they are only using CPTC dataset

Reply:
I have created an issue for this dataset part, you might check it and comment if necessary. I will resolve all other comments related to this issue